src/commands/tui/widgets/text_input.rs

```diff
@@ -40,22 +50,29 @@ impl ProcessEvent for TextInput {
     type Result = TextInputResult;
     fn input(&mut self, event: Event) -> TextInputResult {
         if let Event::Key(key) = event {
+            let KeyEvent {
+                code, modifiers, ..
+            } = key;
             use KeyCode::*;
-            match key {
```

Comment:
the `changeable` parameter has heavy consequences on input interpretation.

For this reason, we could want to have completely separated `match (code, modifiers)`  for `changeable` and `!changeable` cases

Reply:
Good idea! I made two blocks how to handle keys.